### ISSUE TITLE
`renderLayer` called on _every_ update?

### ISSUE THREAD

**Issue by lydell** [developit/preact-portal#17]
The point of `<Portal>` is to pass children to it (that are rendered somewhere else). And `props.children !== prevProps.children` always, right? If so, doesn't this mean that the for-if check in `componentDidUpdate` _always_ triggers?

https://github.com/developit/preact-portal/blob/ba212795be780de0d0a0c9725c69b1998e6f1123/src/preact-portal.js#L10-L16

**Comment by developit** [comment]
This is true of any Preact/react component. Update Cascade can be mitigated using shouldComponentUpdate or memo. 

Fwiw the render() call diffs, so what's happening here isn't necessarily a DOM mutation, but rather the flow of data through the virtual DOM tree.

**Comment by lydell** [comment]
Ok! But the `for-if` loop does not do anything here, right? So it could be removed?

```js
 componentDidUpdate(props) { 
 	return setTimeout(this.renderLayer); 
} 
```

**Comment by developit** [comment]
Technically it's possible to avoid triggering rerenders by hoisting things out of render:

```js
const myPortal = <Portal into="body"><SomeComponent /></Portal>

class Outer {
  render() {
    return <div>{myPortal}</div>
  }
}
```

**Comment by lydell** [comment]
Thanks for the suggestion! I'm not having a performance problem or anything though, I'm just trying to understand how these things work:

Here's a codesandbox showing what I mean: https://codesandbox.io/s/k54oz8mzm3?fontsize=14

I the demo, the `<Portally>` component _always_ says that `props.children` change (even though they don't really, kind of).
